{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.7.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"!git clone https://github.com/s-nlp/detox","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:04:34.570224Z","iopub.execute_input":"2023-03-26T14:04:34.570792Z","iopub.status.idle":"2023-03-26T14:04:39.004486Z","shell.execute_reply.started":"2023-03-26T14:04:34.570742Z","shell.execute_reply":"2023-03-26T14:04:39.002601Z"},"trusted":true},"execution_count":1,"outputs":[{"name":"stdout","text":"Cloning into 'detox'...\nremote: Enumerating objects: 180, done.\u001b[K\nremote: Counting objects: 100% (180/180), done.\u001b[K\nremote: Compressing objects: 100% (158/158), done.\u001b[K\nremote: Total 180 (delta 50), reused 118 (delta 14), pack-reused 0\u001b[K\nReceiving objects: 100% (180/180), 17.92 MiB | 9.76 MiB/s, done.\nResolving deltas: 100% (50/50), done.\n","output_type":"stream"}]},{"cell_type":"code","source":"!pip install -r detox/requirements.txt --quiet","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:04:41.826182Z","iopub.execute_input":"2023-03-26T14:04:41.826649Z","iopub.status.idle":"2023-03-26T14:05:38.283004Z","shell.execute_reply.started":"2023-03-26T14:04:41.826605Z","shell.execute_reply":"2023-03-26T14:05:38.281747Z"},"trusted":true},"execution_count":2,"outputs":[{"name":"stdout","text":"\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\nvirtualenv 20.17.1 requires importlib-metadata>=4.8.3; python_version < \"3.8\", but you have importlib-metadata 3.10.1 which is incompatible.\npynndescent 0.5.8 requires importlib-metadata>=4.8.1; python_version < \"3.8\", but you have importlib-metadata 3.10.1 which is incompatible.\nmarkdown 3.4.1 requires importlib-metadata>=4.4; python_version < \"3.10\", but you have importlib-metadata 3.10.1 which is incompatible.\nlightning-utilities 0.7.1 requires importlib-metadata>=4.0.0; python_version < \"3.8\", but you have importlib-metadata 3.10.1 which is incompatible.\nlibrosa 0.10.0 requires soundfile>=0.12.1, but you have soundfile 0.11.0 which is incompatible.\nibis-framework 2.1.1 requires importlib-metadata<5,>=4; python_version < \"3.8\", but you have importlib-metadata 3.10.1 which is incompatible.\ngym 0.23.1 requires importlib-metadata>=4.10.0; python_version < \"3.10\", but you have importlib-metadata 3.10.1 which is incompatible.\ngoogle-cloud-pubsublite 1.6.0 requires overrides<7.0.0,>=6.0.1, but you have overrides 3.1.0 which is incompatible.\ncmudict 1.0.13 requires importlib-metadata<6.0.0,>=5.1.0, but you have importlib-metadata 3.10.1 which is incompatible.\u001b[0m\u001b[31m\n\u001b[0m\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0m","output_type":"stream"}]},{"cell_type":"code","source":"!pip install fairseq==0.10.0 --quiet","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:05:43.024804Z","iopub.execute_input":"2023-03-26T14:05:43.025445Z","iopub.status.idle":"2023-03-26T14:06:45.770745Z","shell.execute_reply.started":"2023-03-26T14:05:43.025387Z","shell.execute_reply":"2023-03-26T14:06:45.769411Z"},"trusted":true},"execution_count":3,"outputs":[{"name":"stdout","text":"\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0m","output_type":"stream"}]},{"cell_type":"code","source":"!mkdir -p models","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:03.249597Z","iopub.execute_input":"2023-03-26T14:07:03.250194Z","iopub.status.idle":"2023-03-26T14:07:04.347438Z","shell.execute_reply.started":"2023-03-26T14:07:03.250145Z","shell.execute_reply":"2023-03-26T14:07:04.345197Z"},"trusted":true},"execution_count":4,"outputs":[]},{"cell_type":"code","source":"!mkdir -p models/cola","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:06.303216Z","iopub.execute_input":"2023-03-26T14:07:06.304610Z","iopub.status.idle":"2023-03-26T14:07:07.399075Z","shell.execute_reply.started":"2023-03-26T14:07:06.304525Z","shell.execute_reply":"2023-03-26T14:07:07.396943Z"},"trusted":true},"execution_count":5,"outputs":[]},{"cell_type":"code","source":"!mkdir -p models/cola/cola-bin","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:09.058761Z","iopub.execute_input":"2023-03-26T14:07:09.059236Z","iopub.status.idle":"2023-03-26T14:07:10.169107Z","shell.execute_reply.started":"2023-03-26T14:07:09.059184Z","shell.execute_reply":"2023-03-26T14:07:10.167048Z"},"trusted":true},"execution_count":6,"outputs":[]},{"cell_type":"code","source":"!gdown https://drive.google.com/drive/folders/18G5ZfLRKTMlV0Ke4shztUlGZ6H8j8AD6 -O  models/cola/cola-bin/input0","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:11.872516Z","iopub.execute_input":"2023-03-26T14:07:11.873022Z","iopub.status.idle":"2023-03-26T14:07:14.216275Z","shell.execute_reply.started":"2023-03-26T14:07:11.872971Z","shell.execute_reply":"2023-03-26T14:07:14.214851Z"},"trusted":true},"execution_count":7,"outputs":[{"name":"stdout","text":"/opt/conda/lib/python3.7/site-packages/gdown/parse_url.py:31: UserWarning: You specified Google Drive Link but it is not the correct link to download the file. Maybe you should try: https://drive.google.com/uc?id=None\n  url=\"https://drive.google.com/uc?id={}\".format(file_id)\nDownloading...\nFrom: https://drive.google.com/drive/folders/18G5ZfLRKTMlV0Ke4shztUlGZ6H8j8AD6\nTo: /kaggle/working/models/cola/cola-bin/input0\n255kB [00:00, 1.14MB/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"!gdown https://drive.google.com/drive/folders/1UlU0g9HUerK0xW8B_MKV08DkIPEs44-Q -O  models/cola/cola-bin/label","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:16.255584Z","iopub.execute_input":"2023-03-26T14:07:16.256052Z","iopub.status.idle":"2023-03-26T14:07:18.642540Z","shell.execute_reply.started":"2023-03-26T14:07:16.256009Z","shell.execute_reply":"2023-03-26T14:07:18.640876Z"},"trusted":true},"execution_count":8,"outputs":[{"name":"stdout","text":"/opt/conda/lib/python3.7/site-packages/gdown/parse_url.py:31: UserWarning: You specified Google Drive Link but it is not the correct link to download the file. Maybe you should try: https://drive.google.com/uc?id=None\n  url=\"https://drive.google.com/uc?id={}\".format(file_id)\nDownloading...\nFrom: https://drive.google.com/drive/folders/1UlU0g9HUerK0xW8B_MKV08DkIPEs44-Q\nTo: /kaggle/working/models/cola/cola-bin/label\n256kB [00:00, 957kB/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"!gdown https://drive.google.com/1rBSrbL_6gfDqOCpNz1JrOJK9gKrkKxhv -O models/cola/checkpoint_best.pt","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:21.525319Z","iopub.execute_input":"2023-03-26T14:07:21.526434Z","iopub.status.idle":"2023-03-26T14:07:23.462673Z","shell.execute_reply.started":"2023-03-26T14:07:21.526373Z","shell.execute_reply":"2023-03-26T14:07:23.461041Z"},"trusted":true},"execution_count":9,"outputs":[{"name":"stdout","text":"/opt/conda/lib/python3.7/site-packages/gdown/parse_url.py:31: UserWarning: You specified Google Drive Link but it is not the correct link to download the file. Maybe you should try: https://drive.google.com/uc?id=None\n  url=\"https://drive.google.com/uc?id={}\".format(file_id)\nDownloading...\nFrom: https://drive.google.com/1rBSrbL_6gfDqOCpNz1JrOJK9gKrkKxhv\nTo: /kaggle/working/models/cola/checkpoint_best.pt\n100%|██████████████████████████████████████| 96.6k/96.6k [00:00<00:00, 46.8MB/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"!gdown https://drive.google.com/drive/folders/1lBN2nbzxtpqbPUyeURtzt0k1kBY6u6Mj -O models/sim","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:25.588947Z","iopub.execute_input":"2023-03-26T14:07:25.589407Z","iopub.status.idle":"2023-03-26T14:07:28.278843Z","shell.execute_reply.started":"2023-03-26T14:07:25.589365Z","shell.execute_reply":"2023-03-26T14:07:28.277304Z"},"trusted":true},"execution_count":10,"outputs":[{"name":"stdout","text":"/opt/conda/lib/python3.7/site-packages/gdown/parse_url.py:31: UserWarning: You specified Google Drive Link but it is not the correct link to download the file. Maybe you should try: https://drive.google.com/uc?id=None\n  url=\"https://drive.google.com/uc?id={}\".format(file_id)\nDownloading...\nFrom: https://drive.google.com/drive/folders/1lBN2nbzxtpqbPUyeURtzt0k1kBY6u6Mj\nTo: /kaggle/working/models/sim\n235kB [00:00, 1.50MB/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"!gdown https://drive.google.com/uc?id=1XPhNteM9yvArWrt_eZ7HOq8GaK2trYlq","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:30.764011Z","iopub.execute_input":"2023-03-26T14:07:30.764492Z","iopub.status.idle":"2023-03-26T14:07:33.613705Z","shell.execute_reply.started":"2023-03-26T14:07:30.764450Z","shell.execute_reply":"2023-03-26T14:07:33.612021Z"},"trusted":true},"execution_count":11,"outputs":[{"name":"stdout","text":"Downloading...\nFrom: https://drive.google.com/uc?id=1XPhNteM9yvArWrt_eZ7HOq8GaK2trYlq\nTo: /kaggle/working/test_hard\n100%|████████████████████████████████████████| 141k/141k [00:00<00:00, 43.1MB/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"!gdown https://drive.google.com/uc?id=1zLuJ51Nrt_wDuVzqW1MIfHwb9K3RvemY","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:07:35.436159Z","iopub.execute_input":"2023-03-26T14:07:35.436891Z","iopub.status.idle":"2023-03-26T14:07:37.908009Z","shell.execute_reply.started":"2023-03-26T14:07:35.436842Z","shell.execute_reply":"2023-03-26T14:07:37.906646Z"},"trusted":true},"execution_count":12,"outputs":[{"name":"stdout","text":"Downloading...\nFrom: https://drive.google.com/uc?id=1zLuJ51Nrt_wDuVzqW1MIfHwb9K3RvemY\nTo: /kaggle/working/model_outputs\n100%|████████████████████████████████████████| 138k/138k [00:00<00:00, 61.9MB/s]\n","output_type":"stream"}]},{"cell_type":"code","source":"! python \"detox/emnlp2021/metric/metric.py\" \\\n--inputs \"/kaggle/working/test_hard\" \\\n--preds \"/kaggle/working/model_outputs\" \\\n--cola_classifier_path \"/models/cola\"","metadata":{"execution":{"iopub.status.busy":"2023-03-26T14:33:57.289055Z","iopub.execute_input":"2023-03-26T14:33:57.289587Z","iopub.status.idle":"2023-03-26T14:34:20.766490Z","shell.execute_reply.started":"2023-03-26T14:33:57.289542Z","shell.execute_reply":"2023-03-26T14:34:20.765365Z"},"trusted":true},"execution_count":15,"outputs":[{"name":"stdout","text":"Calculating style of predictions\nTraceback (most recent call last):\n  File \"detox/emnlp2021/metric/metric.py\", line 161, in <module>\n    accuracy_by_sent = classify_preds(args, preds)\n  File \"detox/emnlp2021/metric/metric.py\", line 33, in classify_preds\n    tokenizer = RobertaTokenizer.from_pretrained(model_name)\n  File \"/opt/conda/lib/python3.7/site-packages/transformers/tokenization_utils_base.py\", line 1700, in from_pretrained\n    user_agent=user_agent,\n  File \"/opt/conda/lib/python3.7/site-packages/transformers/file_utils.py\", line 1410, in cached_path\n    local_files_only=local_files_only,\n  File \"/opt/conda/lib/python3.7/site-packages/transformers/file_utils.py\", line 1581, in get_from_cache\n    \"Distant resource does not have an ETag, we won't be able to reliably ensure reproducibility.\"\nOSError: Distant resource does not have an ETag, we won't be able to reliably ensure reproducibility.\n","output_type":"stream"}]}]}